spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/mail/MailSenderAutoConfiguration.java

```diff
@@ -40,6 +41,7 @@
  */
 @Configuration
 @ConditionalOnClass({MailSender.class, AmazonSimpleEmailService.class})
+@ConditionalOnMissingBean(MailSender.class)
```

Comment:
`org.springframework.boot.autoconfigure.mail.MailSenderAutoConfiguration.class` will only do something if the configuration properties `spring.mail.host` or `spring.mail.jndi-name` are set, see https://github.com/spring-projects/spring-boot/blob/v1.4.2.RELEASE/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mail/MailSenderAutoConfiguration.java#L105 I think if a user set those properties, then they want to configure mail and not rely on SES.

So actually, I think doing the opposite of what you're suggesting the is the way to go, add:
`@AutoConfigureAfter(org.springframework.boot.autoconfigure.mail.MailSenderAutoConfiguration.class)`